stm/src/boxedalloc.rs

```diff
@@ -0,0 +1,116 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! # Compatibility Module
+//!  
+//! This is a compatibility module, that must be replace with the upcoming memory
+//! functionality for stronghold
+
+use std::sync::{Arc, Mutex};
+use thiserror::Error as DeriveError;
+use zeroize::Zeroize;
+
+pub enum MemoryConfiguration {}
+
+pub struct GuardedMemory<T> {
+    pub(crate) inner: T,
+}
+
+#[derive(DeriveError, Debug)]
+pub enum MemoryError {
+    #[error("Out of memory {0}")]
+    OutOfMemory(String),
+}
+
+#[derive(DeriveError, Debug)]
+pub enum KeyError {
+    #[error("Could not derive key")]
+    DerivationError,
+}
+
+/// The [`CompositeKey`] implements
+/// a keying scheme, that fragments an initial key
+/// and reconstructs the key on demand.
+///
+/// This impl is by no means the intended impl, but shall
+/// provide minimal functionality
+pub struct CompositeKey<K>
+where
+    K: AsRef<Vec<u8>> + Zeroize + Clone,
+{
+    x: Arc<Mutex<K>>,
+    y: Arc<Mutex<K>>,
+}
+
+impl<K> CompositeKey<K>
+where
+    K: AsRef<Vec<u8>> + Zeroize + Clone,
+{
+    /// Initializes the internal key fragments, consume the
+    /// provided actual key. The key must have a specific size,
+    /// depending on the underlying hashing algorithm.
+    pub fn new(key: K) -> Result<Self, KeyError> {
+        Ok(Self {
+            x: Arc::new(Mutex::new(key.clone())),
```

Comment:
Since this is a specific sized key type, I wonder can we replace the vector with a fixed size slice and avoid the clone here?  Maybe use the new rust sized generic syntax to make something very easy to use.  I know its just an stand in at this point, just curious.

Reply:
True that, and this part will definitely not land in the final version. The `boxedalloc` module shall and will be superseded by the upcoming [`runtime` changes ](https://github.com/iotaledger/stronghold.rs/pull/316). This module is just a stub to allow implementation on interface level at least. Sorry, I should have made that clear :sweat_smile: